### ISSUE TITLE
@angular-eslint/no-uncalled-signals does not detect signal-based queries (e.g., viewChild)

### ISSUE THREAD

**Issue by venkatesh-nagineni** [angular-eslint/angular-eslint#2904]
The ESLint rule @angular-eslint/no-uncalled-signals does not trigger when using signal-based queries such as viewChild() without invoking them (i.e., missing ())

This appears to be a gap in the rule's detection logic.

reproduction:

```
import { Component, viewChild } from '@angular/core';

@Component({
  selector: 'app-test',
  template: `<div #el></div>`,
})
export class TestComponent {
  el = viewChild('el');

  ngAfterViewInit() {
    if (this.el) { // ❌ should be this.el()
      console.log('Element exists');
    }
  }
}
```
